### PR TITLE
Refactor argument passing #2

### PR DESCRIPTION
**Pull request by anthrotype**
continuation of @simoncozens's #533 PR, with some fixes to ensure we don't let through unexpected keyword arguments

### CODE REVIEW

**Comment by anthrotype** [comment]
CLA bot needs @simoncozens' consent

**Comment by behdad** [comment]
Much better. :)

File: Lib/ufo2ft/util.py [hunk 1]
```diff
@@ -487,3 +487,27 @@ def _loadPluginFromString(spec, moduleName, isValidFunc):
 def quantize(number, factor):
     """Round to a multiple of the given parameter"""
     return factor * otRound(number / factor)
+
+
+def init_kwargs(kwargs, defaults):
+    """Initialise kwargs default values.
+
+    Raise TypeError with unexpected keyword arguments (missing from 'defaults').
```
Comment (madig) [inline Lib/ufo2ft/util.py:495]:
```suggestion
    To be used as the first function in top-level `ufo2ft.compile*` functions.

    Raise TypeError with unexpected keyword arguments (missing from 'defaults').
```

File: Lib/ufo2ft/util.py [hunk 2]
```diff
@@ -487,3 +487,27 @@ def _loadPluginFromString(spec, moduleName, isValidFunc):
...
+        func_name = currentframe().f_back.f_code.co_name
+        raise TypeError(
+            f"{func_name}() got unexpected keyword arguments: "
+            f"{', '.join(repr(k) for k in extra_kwargs)}"
+        )
+    return {k: (kwargs[k] if k in kwargs else v) for k, v in defaults.items()}
+
+
+def prune_unknown_kwargs(kwargs, *callables):
+    """Inspect callables and return a new dict skipping any unknown arguments"""
```
Comment (madig) [inline Lib/ufo2ft/util.py:509]:
```suggestion
    """Inspect callables and return a new dict skipping any unknown arguments.
    
    To be used after `init_kwargs` to narrow down arguments for underlying code.
    """
```

**Review by madig — APPROVED** [review]
LGTM

**Pull request merged**